### PR TITLE
Support to display workspace feature configured with wildcard

### PR DESCRIPTION
**Pull request by ruanyl**
### Description

1. In workspace update page, support to display workspace feature configured with wildcard.
2. Fixed an issue which generateWorkspacePermissionError return void
3. Fixed an issue which workspace route validation failed due to missing permission mode in schema
4. Hide feature configure panel for management workspace

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/workspace/public/components/workspace_creator/workspace_form.tsx [hunk 1]
```diff
@@ -324,11 +344,26 @@ export const WorkspaceForm = ({
...
+        formData.features.some((feat) => !defaultFeatures.includes(feat));
+
+      if (!featureConfigChanged) {
+        // If feature config not changed, set workspace feature config to the original value.
+        // The reason why we do this is when a workspace feature is configured by wildcard,
+        // such as `['@management']` or `['*']`. The form value `formData.features` will be
+        // expanded to array of individual feature id, if the feature hasn't changed, we will
+        // set the feature config back to the original value so that category wildcard won't
+        // expanded to feature ids
+        formData.features = defaultValues?.features ?? [];
```
Comment (wanglam) [inline src/plugins/workspace/public/components/workspace_creator/workspace_form.tsx:359]: Is that means the wildcard match will be removed after add some new feature ids?

Comment (ruanyl) [reply src/plugins/workspace/public/components/workspace_creator/workspace_form.tsx:359]: Yes, that makes sense to me as that's what the user selected

**Review by SuZhou-Joe — APPROVED** [review]

File: src/plugins/workspace/public/components/workspace_creator/workspace_form.tsx (hunk 1 above)
Comment (wanglam) [reply src/plugins/workspace/public/components/workspace_creator/workspace_form.tsx:359]: OK. Just wonder if we can support wildcard for the feature category level. Which means if user select a feature category, we will store wildcard feature rule. If they remove some one, we will change to feature ids.

Comment (ruanyl) [reply src/plugins/workspace/public/components/workspace_creator/workspace_form.tsx:359]: Sounds good to me, we could have an improvement ticket for this.

**Review by raintygao — APPROVED** [review]

**Review by wanglam — APPROVED** [review]

**Pull request merged**